base_layer/core/src/consensus/consensus_manager.rs

```diff
@@ -69,17 +98,159 @@ impl ConsensusManager {
 
     /// Returns the genesis block for the selected network.
     #[cfg(feature = "base_node")]
-    pub fn get_genesis_block(&self) -> ChainBlock {
+    pub fn get_genesis_block(&self) -> Result<ChainBlock, ConsensusManagerError> {
         use crate::blocks::genesis_block::get_genesis_block;
         let network = self.inner.network.as_network();
-        match network {
+        let genesis_block = match network {
             Network::LocalNet => self
                 .inner
                 .gen_block
                 .clone()
                 .unwrap_or_else(|| get_genesis_block(network)),
             _ => get_genesis_block(network),
+        };
+        self.check_genesis_block(&genesis_block)?;
```

Comment:
Not going to block this, but not a fan of this. 
This according to me gives us no benefits, all it does is make the boot-up time a node slower by 10-20 seconds.

Reply:
I hear you, and I don't. I am currently fixing many invalid tests, and release mode should be a lot faster. Personally, I like the formal check on the genesis block.